Library/Homebrew/utils.rb

```diff
@@ -130,8 +130,13 @@ def odeprecated(method, replacement = nil, options = {})
   # - Location outside of 'compat/'.
   # - Location of caller of deprecated method (if all else fails).
   backtrace = options.fetch(:caller, caller)
+  tap_message = nil
   caller_message = backtrace.detect do |line|
-    line.start_with?("#{HOMEBREW_LIBRARY}/Taps/")
+    if line =~ %r{#{HOMEBREW_LIBRARY}/Taps/([\w-]+)/([\w-]+)/}
```

Comment:
`[\w-]+` is a reasonable assumption, but I don't think we're making this (still restrictive) assumption anywhere else in the code. For simplicity, I'd make this just `[^/]+`. Also please anchor the regular expression and always escape interpolated strings (you can't guarantee they won't contain characters with special meaning in regular expressions). Everything combined:

```ruby
if line =~ %r{^#{Regexp.escape(HOMEBREW_LIBRARY)}/Taps/([^/]+)/([^/]+)/}
```

You could also further simplify and leave splitting of the argument to `Tap.fetch` by using:

```ruby
if line =~ %r{^#{Regexp.escape(HOMEBREW_LIBRARY)}/Taps/([^/]+/[^/]+)/}
  tap = Tap.fetch($1)
```